src/gwt/src/org/rstudio/core/client/files/filedialog/FileSystemDialog.java

```diff
@@ -190,6 +190,14 @@ protected void accept()
    public void showModal()
    {
       super.showModal();
+      
+      // Modal dialogs always open with zindex of 1001 for the dialog and 1000
+      // for the glass. Therefore, when opening this file dialog inside a modal,
+      // the glass element fails to cover the existing modal. Setting the zIndex
+      // higher than the average modal fixes this issues.
+      getGlassElement().getStyle().setZIndex(2000);
+      getElement().getStyle().setZIndex(2001);
+      
```

Comment:
Interesting! I thought I was the only one opening nested modals, but that was not the case. Yes, you are right, this change hides modal dialogs on top of the file dialog. However, the interesting finding is that all nested modals show this behavior. See example bellow with modals not being modal under the options menu:

<img width="1440" alt="screen shot 2016-01-27 at 11 45 25 am" src="https://cloud.githubusercontent.com/assets/3478847/12626053/7cef8562-c4eb-11e5-824f-390958a308c7.png">

I tracked this back to the original 2010 commit...

https://github.com/rstudio/rstudio/blob/c719a597b1d6192f72d990ef4152b2b92736c725/src/gwt/src/org/rstudio/core/client/theme/res/themeStyles.css

We set a couple z-indexes that override the default GWT styles:

```
.gwt-DialogBox-ModalDialog {
   z-index: 1001;
}

.gwt-SuggestBoxPopup {
   z-index: 1002;	
}
```

This works for single level modals, but for nested one we push the `.gwt-PopupPanelGlass` behind the first-level modal leaving this dialog interactive.

GWT does create the absolute positioned divs sequentially under the `body` tag so there shouldn't be a need to assigned incremental ones. However, we do need to separate modals `zindex` from other components to ensure all modals are above the plane (including `.gwt-PopupPanelGlass` which being kept at `1000` with this change):

```
.gwt-DialogBox-ModalDialog {
   z-index: 1000;
}

.gwt-SuggestBoxPopup {
   z-index: 1000;	
}
```

Here is the same options dialog with this change:

<img width="1440" alt="screen shot 2016-01-27 at 11 51 21 am" src="https://cloud.githubusercontent.com/assets/3478847/12626197/490d0dea-c4ec-11e5-96e7-8dfbb8ff543b.png">